torch/distributed/checkpoint/state_dict.py

```diff
@@ -561,6 +561,19 @@ def _load_model_state_dict(
         )
         for fqn, local_state in local_state_dict.items():
             state_dict[fqn] = local_state
+    elif info.fsdp_context() and info.full_state_dict:
```

Comment:
Why do we care about `fsdp_context()`? This feature should be good for all parallelisms.

Reply:
I was not sure if the error only appears with 'fsdp', will remove it in the next commit.